augmentation/augmentation.hpp

```diff
@@ -0,0 +1,172 @@
+/**
+ * @file augmentation.hpp
+ * @author Kartik Dutt
+ * 
+ * Definition of Augmentation class for augmenting data.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#include <mlpack/methods/ann/layer/bilinear_interpolation.hpp>
+#include <boost/regex.hpp>
+
+#ifndef MODELS_AUGMENTATION_HPP
+#define MODELS_AUGMENTATION_HPP
+
+/**
+ * Augmentation class used to perform augmentations / transform the data.
+ * For the list of supported augmentation, take a look at our wiki page.
+ *
+ * @code
+ * Augmentation<> augmentation({"horizontal-flip", "resize = (224, 224)"}, 0.2);
+ * augmentation.Transform(dataloader.TrainFeatures);
+ * @endcode
+ * 
+ * @tparam DatasetType Datatype on which augmentation will be done.
+ */
+template<typename DatasetType = arma::mat>
```

Comment:
Seems like the class template parameter isn't needed, if we add a template for the method.

Reply:
Ahh, I was doing that earlier but it gave me a strange error with the map [here](https://github.com/mlpack/models/pull/13/files/e97e616a80ee81b2afb8ee746f18663a13c208fc#diff-d8125dd9142e99fd90c585af5f57daa4R161), 
```
std::unordered_map<std::string, void(*)(DatasetType&,
       size_t, size_t, size_t, std::string&)> augmentationMap;
```
So I decided to added a template for the class rather than the method.

Reply:
But looks like the container is used in a single method, so I could be instantiated locally as well? I think removing the class template would make it easier for a user to use the augmentation feature.

Reply:
> I think removing the class template would make it easier for a user to use the augmentation feature.

Agreed, @kartikdutt18 can you try do that in #18?

Reply:
Ahh, makes sense. Will make the change in #18.